ios/browser/api/bookmarks/brave_bookmarks_api.h

```diff
@@ -0,0 +1,113 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_IOS_BROWSER_API_BOOKMARKS_BRAVE_BOOKMARKS_API_H_
+#define BRAVE_IOS_BROWSER_API_BOOKMARKS_BRAVE_BOOKMARKS_API_H_
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+typedef NS_ENUM(NSUInteger, BookmarksNodeType) {
+  URL,
+  FOLDER,
+  BOOKMARK_BAR,
+  OTHER_NODE,
+  MOBILE
+};
+
+typedef NS_ENUM(NSUInteger, BookmarksNodeFaviconState) {
+  INVALID_FAVICON,
+  LOADING_FAVICON,
+  LOADED_FAVICON,
+};
+
+@protocol BookmarkModelObserver;
+@protocol BookmarkModelListener;
+
+OBJC_EXPORT
+@interface BookmarkNode: NSObject
++ (NSString *)kRootNodeGuid;
++ (NSString *)kBookmarkBarNodeGuid;
++ (NSString *)kOtherBookmarksNodeGuid;
++ (NSString *)kMobileBookmarksNodeGuid;
++ (NSString *)kManagedNodeGuid;
+
+- (bool)isPermanentNode;
+- (void)setTitle:(NSString *)title;
+- (NSUInteger)nodeId;
+//- (void)setNodeId:(NSUInteger)id;
+- (NSString *)getGuid;
+- (nullable NSURL *)url;
+- (void)setUrl:(nullable NSURL *)url;
+
+- (nullable NSURL *)iconUrl;
+- (BookmarksNodeType)type;
+- (NSDate *)dateAdded;
+- (void)setDateAdded:(NSDate *)date;
+
+- (NSDate *)dateFolderModified;
+- (void)setDateFolderModified:(NSDate *)date;
+- (bool)isFolder;
+- (bool)isUrl;
+- (bool)isFavIconLoaded;
+- (bool)isFavIconLoading;
+- (bool)isVisible;
```

Comment:
Can't all these methods be readonly properties instead (including the class methods above)